### DISCUSSION TITLE
Can we exclude certain (window) error messages from being shown?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5821] [Ideas]
Hello there,

I have a problem with a vis.js widget I want to create. I get an  `ResizeObserver loop limit exceeded` error and a RSOD pops up, but the message could safely be ignored.

Can we specify error messages that should be ignored / where the RSOD isn't shown?

Thanks :)

**Comment by pmario** [comment]
IMO if it could be safely ignored, the right way would be to tell the vis.js folks to remove it.

**Comment by pmario** [comment]
As far as I know the RSOD catches all JavaScript exceptions and if libraries "throw" an exception, which is not handled by the plugin author. .. So may be there is an option that you catch the exception and resolve it.

**Comment by Jermolene** [comment]
Hi @BurningTreeC this problem has [come up before](https://groups.google.com/g/tiddlywiki/c/vXVLQD58yPQ/m/0f8vPFPtGwAJ). I still don't see a good solution; the point of the RSOD mechanism is to ensure that JS errors don't get hidden in the console because we don't want users to be unaware of errors that might threaten the stability of the system, and thus risk data loss.

**Comment by linonetwo** [comment]
Can we fix this by adding this error handler to the core? https://stackoverflow.com/a/64197640/4617295

I think this is still an issue in the 5.2.2 prerelease.

**Reply by saqimtiaz** [reply]
@linonetwo the problem is that there can be valid scenarios in which we do want to expose the error or at least log it to the console. For example, I ran into this issue in one of my widgets that implements a ResizeObserver and it pointed out a flaw in my code that I may not have otherwise caught.

What is frustrating is that there is no way to discern whether the error arises from a ResizeObserver implemented by TiddlyWiki or the browser itself.

If your use case is embedded video, do try this potential CSS workaround:
```
   video::-webkit-media-controls-timeline {
      align-self: center;
      width: calc(100% - 64px);
    }
```

**Comment by pmario** [comment]
@BurningTreeC Have you seen this "workaround" https://github.com/WICG/resize-observer/issues/38#issuecomment-532833484

**Comment by linonetwo** [comment]
<img width="1195" alt="截屏2022-02-22 下午4 31 01" src="https://user-images.githubusercontent.com/3746270/155092910-edde71cf-221f-4fbf-933d-c7491046671a.png">
<img width="845" alt="截屏2022-02-22 下午4 31 41" src="https://user-images.githubusercontent.com/3746270/155092999-b7a119fb-ea19-4ffe-bc34-f5cf16ef9ed4.png">

@saqimtiaz seems not working, after restart of wiki, did that work for you?

**Reply by linonetwo** [reply]
I'm embedding video via transclusion https://talk.tiddlywiki.org/t/rfc-generic-form-handling/2468

**Reply by saqimtiaz** [reply]
Try playing with the width in the CSS expression. The underlying problem is that the video controls trigger a resize observer.

**Reply by saqimtiaz** [reply]
@linonetwo it does work for me. However it might be that the CSS needs to be adjusted depending on the video resolution.

You can also hide the video controls via CSS:

```
video::-webkit-media-controls { 
      display: none;
}
```

**Reply by linonetwo** [reply]
I'm using

```css
video::-webkit-media-controls-timeline {
  width: 200px;
}
```

So I can still control the timeline...Thanks.

**Comment by pmario** [comment]
There seems to be a workaround, which imo may be better than blocking the RSOD completely. https://github.com/WICG/resize-observer/issues/38#issuecomment-422085926

```
const observer = new ResizeObserver(()=> {
    setTimeout(() => updateStyleOf(siblingOfObservedElement));
});
```
